### PR TITLE
Migrate article about opensearch replication factors

### PR DESCRIPTION
**Pull request by anelook**
# What changed, and why it matters

Migration of https://help.aiven.io/en/articles/2342117-automatic-adjustment-of-elasticsearch-index-replication-factors

Note, advanced property disable_replication_factor_adjustment is deprecated and in dev-sandbox / dev-advocates no longer available. That's why I excluded the part where we recommend setting this value.

### CODE REVIEW

**Comment by lornajane** [comment]
See also: #357

File: docs/products/opensearch/reference/index-replication.rst [hunk 1]
```diff
@@ -0,0 +1,36 @@
+Automatic adjustment of replication factors
+===========================================
+
+Aiven for OpenSearch automatically adjusts index replication factors
+to ensure data availability and service functionality.
+
+When replication factor is automatically adjusted?
```
Comment (TibsAtWork) [inline docs/products/opensearch/reference/index-replication.rst:7]: It was like this in the original, but if I was writing this from scratch, I'd put the "`is`" after the "`When`" - that is, "`When is ...`"

File: docs/products/opensearch/reference/index-replication.rst [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+-  If ``number_of_replicas`` is too large for current cluster size, it
+   is automatically lowered to maximum possible value (number of nodes
+   on the cluster - 1).
+
+-  If ``number_of_replicas`` is 0 on a multi-node cluster, it is
+   automatically increased to 1.
+
+-  If ``number_of_replicas`` is anywhere between 1 and maximum value, it
```
Comment (TibsAtWork) [inline docs/products/opensearch/reference/index-replication.rst:17]: Minor point, but since we're editing stuff, "`the maximum`"

File: docs/products/opensearch/reference/index-replication.rst [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+Aiven for OpenSearch automatically adjusts index replication factors
+to ensure data availability and service functionality.
+
+When replication factor is automatically adjusted?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+-  If ``number_of_replicas`` is too large for current cluster size, it
+   is automatically lowered to maximum possible value (number of nodes
+   on the cluster - 1).
```
Comment (TibsAtWork) [inline docs/products/opensearch/reference/index-replication.rst:12]: I would rephrase as something like:
```
The maximum value for ``number_of_replicas`` is the number of nodes in the cluster - 1. 
```
and then carry on with the example from below:
```
For example, for a 3-node cluster, the maximum ``number_of_replicas`` is 2, which means all shards on the index are replicated to all 3 nodes.
```
and finish with
```
If ``number_of_replicas`` is too large, it will automatically be reduced to the maximum value.
```

File: docs/products/opensearch/reference/index-replication.rst [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+Lowering replication factor
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When replication factor ( ``number_of_replicas`` ) is set to larger than
+size of the cluster, ``number_of_replicas`` is automatically lowered, as
+it is not possible to replicate indexes (shards) to a larger number of
+nodes than what exists on the cluster. Do note number replication factor
+is ``number_of_replicas`` + 1. For example, for three-node cluster,
+maximum ``number_of_replicas`` is 2, which means all shards on the index
+are replicated to all three nodes.
```
Comment (TibsAtWork) [inline docs/products/opensearch/reference/index-replication.rst:29]: If the example is moved earlier, then I think this is all said by the previous section, so can be removed.

File: docs/products/opensearch/reference/index-replication.rst [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+Increasing replication factor
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+For multi-node clusters, Aiven for OpenSearch automatically increases
+``number_of_replicas`` to 1, if it is set to 0. This is to ensure no
+data-loss occurs if one node is lost.
```
Comment (TibsAtWork) [inline docs/products/opensearch/reference/index-replication.rst:36]: Again, this was already said in the list in the earlier section.

**Review by TibsAtWork — COMMENTED** [review]
I *think* the original text repeated itself a bit much, so it could be shortened.

Unless you feel there's value in saying the max / min points in different ways, so people don't miss them.
